### ISSUE TITLE
Add support for C's default argument promotion

### ISSUE THREAD

**Issue by ekrich** [scala-native/scala-native#721]
Example code:
```
import scalanative.native._
import stdio._

object Tester {
  def main(args: Array[String]): Unit = {
    val ptr = stackalloc[CStruct2[CFloat, CFloat]]
    !ptr._1 = 3.0f
    !ptr._2 = 4.0f
    printf(c"c -> first %f second %f\n", !ptr._1, !ptr._2)
    println(s"s -> first ${!ptr._1} second ${!ptr._2}")
  }
}
```
Output:
```
s -> first 3.000000 second 4.000000
c -> first 0.000000 second 0.000000
```

**Comment by jonas** [comment]
`float` values needs to be cast to `double` before passed to `printf` according to [this stackoverflow question](http://stackoverflow.com/questions/4264127/correct-format-specifier-for-double-in-printf).

This C code:
```c
  struct s { float a; float b; } s = { 3.0f, 4.0f };
  struct s *ptr = &s;

  printf("c -> first %f second %f\n", ptr->a, ptr->b);
```
is compiled to
```ll
  %6 = getelementptr inbounds %struct.s, %struct.s* %5, i32 0, i32 0
  %7 = load float, float* %6, align 4
  %8 = fpext float %7 to double
  %9 = load %struct.s*, %struct.s** %3, align 8
  %10 = getelementptr inbounds %struct.s, %struct.s* %9, i32 0, i32 1
  %11 = load float, float* %10, align 4
  %12 = fpext float %11 to double
  %13 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([25 x i8], [25 x i8]* @.str, i32 0, i32 0), double %8, double %12)
```
The example code is compiled to:
```ll
  %gen.7 = bitcast i8* %src.3 to { float, float }*
  %gen.8 = getelementptr { float, float }, { float, float }* %gen.7, i32 0, i32 0
  %src.13 = bitcast float* %gen.8 to i8*
  %gen.9 = bitcast i8* %src.13 to float*
  %src.14 = load float, float* %gen.9
  %gen.10 = bitcast i8* %src.3 to { float, float }*
  %gen.11 = getelementptr { float, float }, { float, float }* %gen.10, i32 0, i32 1
  %src.17 = bitcast float* %gen.11 to i8*
  %gen.12 = bitcast i8* %src.17 to float*
  %src.18 = load float, float* %gen.12
  %src.21 = call i32 (i8*, ...) @"printf"(i8* bitcast ([25 x i8]* @"__const:: 770971583.0" to i8*), float %src.14, float %src.18)
```

**Comment by densh** [comment]
That's a bug, thanks for discovering it!

**Comment by densh** [comment]
Fixed in #1426
